unittests/currency_tests.cpp

```diff
@@ -18,7 +18,7 @@ using namespace eosio::chain;
 using namespace eosio::testing;
 using namespace fc;
 
-class currency_tester : public validating_tester {
+class currency_tester : public validating_tester_no_disable_deferrd_trx {
```

Comment:
All the currency_tests except `test_deferred_failure` should work fine even with deferred transactions disabled.

I would rather see that one test case handled as a special case (maybe my making the `currency_tester` a template?) so that most of the test cases can run in the proper environment we expect the network to eventually move to which is disabling the deferred transactions.

Reply:
Thanks. Agreed. I was planning to look into further today to adapt those tests which had to use `validating_tester_no_disable_deferred_trx` such that both before and after work.

Reply:
Made the changes. One question: I tried to add a new test for test_proxy which sets delay to 0 and uses `full` policy. Somehow the transfer did not happen.

Reply:
The proxy contract will use deferred transactions regardless of whether the delay is set to 0 or not. So it makes sense that the deferred transaction would fail to execute in this branch, regardless of whether the disable deferred transaction protocol features are activated or not.

Reply:
Thanks for your explanation!